### ISSUE TITLE
Space key and Enter key are the same for `test` function in v3.1

### ISSUE THREAD

**Issue by c02y** [fish-shell/fish-shell#6746]
~~~
$ fish --version
fish, version 3.1.0

$ uname -a
Linux chz-pc 5.5.8-1-MANJAR

$ echo $TERM
screen-256color
~~~

The following result is for  `sh -c 'env HOME=$(mktemp -d) fish'`

A simple function:
~~~
function read_confirm
  while true
    read -n 1 -p 'echo "Space or Enter? "' -l confirm
    if test "$confirm" = ''
        echo Enter
    else if test "$confirm" = ' '
        echo Space
    end
  end
end
~~~

When using fish v3.0.2
~~~
chz@chz-pc ~> read_confirm
Space or Enter?  # type Space here
Space # <--
Space or Enter?  # type Enter here
Enter
Space or Enter?
~~~
When using fish v3.1.0 (both versions are installed from Manjaro official repo)
~~~
chz@chz-pc ~> read_confirm
Space or Enter?  # type Space here
Enter # <--
Space or Enter?  # type Enter here
Enter
Space or Enter?
~~~

Since I defined several functions to handle different situations for Space key and Enter key using `read  + if test` expression, so those functions now contain bugs for me.

Is it defined like this in fish v3.1 or a bug?

**Comment by faho** [comment]
Duplicate of #6650

**Comment by faho** [comment]
This is caused by `read` trimming whitespace, which was the underlying issue for #6650 and is fixed there by 0c74ff4209b9685de6fda593d391cae1687a6788.

You'll notice with `string escape -- $confirm` that both of these just set $confirm to the empty string in 3.1.0.
